nitransforms/io.py

```diff
@@ -0,0 +1,263 @@
+        sa['m_L'] = val
+        lines = lines[4:]
+        assert lines.pop(0) == 'src volume info'
+        sa['src'] = np.asanyarray(VolumeGeometry.from_string('\n'.join(lines[:8])))
+        lines = lines[8:]
+        assert lines.pop(0) == 'dst volume info'
+        sa['dst'] = np.asanyarray(VolumeGeometry.from_string('\n'.join(lines)))
+        return lt
+
+
+class LinearTransformArray(StringBasedStruct):
+    template_dtype = np.dtype([
+        ('type', 'i4'),
+        ('nxforms', 'i4'),
+        ('subject', 'U1024'),
+        ('fscale', 'f4')])
+    dtype = template_dtype
+    _xforms = None
+
+    def __init__(self,
+                 binaryblock=None,
+                 endianness=None,
+                 check=True):
+        super(LinearTransformArray, self).__init__(binaryblock, endianness, check)
+        self._xforms = [LinearTransform()
+                        for _ in range(self.structarr['nxforms'])]
+
+    def __getitem__(self, idx):
+        if idx == 'xforms':
+            return self._xforms
+        if idx == 'nxforms':
+            return len(self._xforms)
+        return super(LinearTransformArray, self).__getitem__(idx)
+
+    def to_string(self):
+        code = int(self['type'])
+        header = [
+            'type      = {} # {}'.format(code, transform_codes.label[code]),
+            'nxforms   = {}'.format(self['nxforms'])]
+        xforms = [xfm.to_string() for xfm in self._xforms]
+        footer = [
+            'subject {}'.format(self['subject']),
+            'fscale {:.6f}'.format(float(self['fscale']))]
+        return '\n'.join(header + xforms + footer)
+
+    @classmethod
+    def from_string(klass, string):
+        lta = klass()
+        sa = lta.structarr
+        lines = string.splitlines()
+        for key in ('type', 'nxforms'):
+            label, valstring = lines.pop(0).split(' = ')
+            assert label.strip() == key
+
+            val = np.genfromtxt([valstring.encode()],
+                                dtype=klass.dtype[key])
+            sa[key] = val.reshape(sa[key].shape) if val.size else ''
+        for _ in range(sa['nxforms']):
+            lta._xforms.append(
+                LinearTransform.from_string('\n'.join(lines[:25])))
+            lines = lines[25:]
+        if lines:
+            for key in ('subject', 'fscale'):
+                # Optional keys
+                if not lines[0].startswith(key):
+                    continue
+                label, valstring = lines.pop(0).split(' ')
+                assert label.strip() == key
+
+                val = np.genfromtxt([valstring.encode()],
+                                    dtype=klass.dtype[key])
+                sa[key] = val.reshape(sa[key].shape) if val.size else ''
+
+        assert len(lta._xforms) == sa['nxforms']
+        return lta
+
+    @classmethod
+    def from_fileobj(klass, fileobj, check=True):
+        return klass.from_string(fileobj.read())
+
+    def as_type(self, target):
```

Comment:
For me, `as_type()` reads that it would return a modified version of the data without modifying the source data.

```suggestion
    def set_type(self, target):
```